### DISCUSSION TITLE
Embedding python into a Rust Hello World

### DISCUSSION THREAD

**Discussion by lllama** [indygreg/PyOxidizer#652] [Q&A]
My aim is to embed python into a Rust app and have the Rust app manage its lifecycle. Ideally, I'd like the python interpreter to run in a separate process/thread and communicate back with the main rust program, and I'd like to include some of my custom python code. I'm assuming this is all possible.

With that in mind, I've used `cargo new` to create a new Hello World application, and I've added `pyembed` to my `cargo.toml`. But now I'm a little confused as to what the workflow should be. Should I be using `cargo build`, or `pyoxidizer build` to build the file?

**Comment by indygreg** [comment]
Depends on how portable you want your binary to be.

If all you need is something that runs on the machine it was built on, `cargo build` with `pyembed` as a dependency can _just work_. If you want to distribute your application, you probably want to involve `pyoxidizer` to generate linkable build artifacts.

I attempted to document this at https://gregoryszorc.com/docs/pyoxidizer/0.22.0/pyoxidizer_rust_generic_embedding.html. You may want to have a look at the CI jobs that build the `pyoxy` executables, as those do everything you would need to do to build a binary that can be distributed.

**Reply by lllama** [reply]
Awesome! Thanks Greg!
